tests/tt_metal/tt_metal/unit_tests_fast_dispatch/command_queue/test_EnqueueTrace.cpp

```diff
@@ -590,3 +593,128 @@ TEST_F(RandomProgramTraceFixture, TensixTestLargeProgramInBetweenFiveSmallProgra
     Finish(this->device_->command_queue());
     ReleaseTrace(this->device_, trace_id);
 }
+
+TEST_F(RandomProgramTraceFixture, TensixTestProgramsTraceAndNoTrace) {
+    for (uint32_t i = 0; i < NUM_PROGRAMS; i++) {
+        if (i % 10 == 0) {
+            log_info(tt::LogTest, "Creating Program {}", i);
+        }
+        this->programs[i] = CreateProgram();
+        Program& program = this->programs[i];
+        this->create_kernel(program, CoreType::WORKER);
+    }
+
+    std::vector<uint32_t> trace_ids;
+    for (Program& program : this->programs) {
+        const bool use_trace = (rand() % 2) == 0;
+        if (use_trace) {
+            EnqueueProgram(this->device_->command_queue(), program, false);
```

Comment:
it would be better (I think) if we can generate the traces ahead of time then just call EnqueueTrace in the loop at execution time